src/state/quick-bar-mixin.ts

```diff
@@ -15,9 +15,36 @@ const isMacOS = /(Mac|iPhone|iPod|iPad)/i.test(navigator.platform);
 
 export default <T extends Constructor<HassElement>>(superClass: T) =>
   class extends superClass {
+    private isAdmin = false;
+
+    private haveCheckedAdminStatus = false;
+
+    protected updated(changedProps: PropertyValues) {
+      if (this.haveCheckedAdminStatus) {
+        return;
+      }
```

Comment:
I searched high and low for a reliable lifecycle hook that would *only* fire when the `this.hass.user` object was defined, but this is the only one that actually worked.

`firstUpdated` doesn't have user object at this point in bootstrapping.
`hassConnected` just tells us when hass object is available, but still doesn't guarantee a user object
`subscribeFrontendUserData()` seemed promising until I saw that it only fires when "show_advanced" is available.

So I just subscribed to all events and used a 2nd boolean to make sure that hass object updates don't do any extra work.

My worry is that with quick-bar-mixin being attached to such a high-level element, a lot of unnecessary checks will occur even though we've already done the work of attaching keyboard listeners.

Any advice?

Reply:
You are never un registering the shortcut if the user changes. Idk if that's possible without reconnecting and if its not then that's much easier.

Create a `shouldUpdate` return true only if `private _isRegestered = false` is false and if hass.user !== oldHass.user`

This way you arent updating all the time on every hass change